### PR TITLE
docs: update cell

### PR DESCRIPTION
**Pull request by zero0Halo**
**Before submitting a pull request,** please make sure the following is done:

I have done **all** of the following:

- [ ] Added a top level class to all my components `'.anchor-[COMPONENT NAME]'`.
- [x] Used [conventional commits](https://www.conventionalcommits.org) for all work.
- [x] Tested my solution on Mobile & Tablet.
- [ ] Wrote [unit tests](https://jestjs.io/docs/en/getting-started) for states and all behavior (`npm test`) and passed coverage thresholds.
- [ ] Updated snapshots for all permutations (`npm test -- -u`).
- [ ] Accounted for hover, focus, blur, visited, & error states because they are not edge cases.
- [ ] Created TODOs for known edge cases.
- [x] Documented all of my changes (inline & doc site).
- [ ] Made sure that all accessibility errors are resolved.
- [ ] Added [stories](https://storybook.js.org/docs/basics/introduction/) with knobs for all possible configurations.
- [x] De-linted and ran [prettier](https://github.com/prettier/prettier) (`npm run pretty`) on my code.
- [ ] Added name to OWNERS file for all new components
- [ ] If adding a new component, add its export to the rollup config
- [ ] package.json version is bumped (if necessary)

---------
**Outline your feature or bug-fix below**

* `Cell` 
  * Adds the `align` and `valign` props to the api table
  * Adds notes that `middle` and `center` will be deprecated
  * Adds note that `Cell` is also exported as a submodule of `Grid`, and that the standalone will be deprecated
* Updated doc site to use `Cell`'s new `valign` and `align` props

### CODE REVIEW

File: docs/src/pages/components/grid.mdx [hunk 1]
```diff
@@ -24,11 +35,11 @@ with default settings.
 
 ```tsx live
 <Grid debug>
-    <Cell width={4}> I'm cell 1.</Cell>
-    <Cell width={4}> I'm cell 2.</Cell>
-    <Cell width={4}> I'm cell 3.</Cell>
-    <Cell width={6}> I'm cell 4.</Cell>
-    <Cell width={6}> I'm cell 5.</Cell>
+    <Grid.Cell width={4}> I'm cell 1.</Grid.Cell>
```
Comment (sharkparty) [inline docs/src/pages/components/grid.mdx:38]: This could also be destructured at the top of the file to prevent this verbosity.

Comment (zero0Halo) [reply docs/src/pages/components/grid.mdx:38]: Variables can't be declared in the active code editor to my knowledge.

**Review by brandonlilly — APPROVED** [review]
